lib/miq_expression/field.rb

```diff
@@ -122,6 +122,10 @@ def arel_attribute
     target.arel_attribute(column)
   end
 
+  def to_a
```

Comment:
I agree with @imtayadeway here. Not sure what I think about the `to_a` to make the extraction of attributes out of fields.

The extraction logic feels very tightly coupled here